helix-core/src/main/java/org/apache/helix/manager/zk/ParticipantManager.java

```diff
@@ -76,9 +77,16 @@
   final StateMachineEngine _stateMachineEngine;
   final LiveInstanceInfoProvider _liveInstanceInfoProvider;
   final List<PreConnectCallback> _preConnectCallbacks;
+  final HelixParticipantProperty _helixParticipantProperty;
 
   public ParticipantManager(HelixManager manager, HelixZkClient zkclient, int sessionTimeout,
       LiveInstanceInfoProvider liveInstanceInfoProvider, List<PreConnectCallback> preConnectCallbacks) {
+    this(manager, zkclient, sessionTimeout, liveInstanceInfoProvider, preConnectCallbacks, null);
```

Comment:
Is passing `null` a safe thing to do? Wouldn't it be better to pass in an empty instance of `HelixParticipantProperty` to avoid potential NPEs?

Reply:
I mark it as deprecated.